dataloader/renderdata.go

```diff
@@ -0,0 +1,38 @@
+package dataloader
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"log"
+	"net/http"
+
+	"../defination"
+	"../interfaces"
+)
+
+type RenderServiceLoader struct {
+	data interfaces.RenderData
+}
+
+func (service *RenderServiceLoader) Render(writer http.ResponseWriter, postWithComment defination.PostWithCommentsResponse, isJSON bool) {
```

Comment:
Nitpick: this function does 2 things (render json or render xml), so can be separated into 2 smaller functions for readability